src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -15,12 +15,38 @@ def __init__(self, name='', wavelength=None):
         self.name = name
         self.wavelength = wavelength
         self.scat_quantity = ""
-        self.on_q = [np.empty(1), np.empty(1)]
-        self.on_tth = [np.empty(1), np.empty(1)]
-        self.on_d = [np.empty(1), np.empty(1)]
+        self.on_q = [np.empty(0), np.empty(0)]
+        self.on_tth = [np.empty(0), np.empty(0)]
+        self.on_d = [np.empty(0), np.empty(0)]
         self._all_arrays = [self.on_q, self.on_tth]
         self.metadata = {}
 
+    def __eq__(self, other):
+        if not isinstance(other, Diffraction_object):
+            return NotImplemented
+
+        attributes_to_test = {"name", "wavelength", "scat_quantity", "on_q", "on_tth", "on_d", "metadata"}
```

Comment:
I am not sure why this code-block is here.  It looks like the test, but we only need it as a test, not as part of the code.  I suggest to delete it.